code/game/machinery/_machinery.dm

```diff
@@ -218,10 +218,17 @@ Class Procs:
 	if(panel_open && !(interaction_flags_machine & INTERACT_MACHINE_OPEN))
 		if(!silicon || !(interaction_flags_machine & INTERACT_MACHINE_OPEN_SILICON))
 			return FALSE
-	if(!silicon && (interaction_flags_machine & INTERACT_MACHINE_REQUIRES_SILICON))
-		return FALSE
-	else if(silicon && !(interaction_flags_machine & INTERACT_MACHINE_ALLOW_SILICON))
-		return FALSE
+
+	if(silicon)
+		if(!(interaction_flags_machine & INTERACT_MACHINE_ALLOW_SILICON))
+			return FALSE
+	else
+		if(interaction_flags_machine & INTERACT_MACHINE_REQUIRES_SILICON)
+			return FALSE
+		if(!Adjacent(user))
+			var/mob/living/carbon/H = user
+			if(istype(H) && H.has_dna() && !H.dna.check_mutation(TK))
+				return FALSE
```

Comment:
If they're not adjacent but they have no dna they can interact?

Reply:
I'm just looking to see if they have the TK power, as in that case they should be able to interact with machines without being adjacent. The `istype(H) && H.has_dna()` are just there to make sure the user has DNA to have checked in the first place. If the user doesn't have DNA they would definitely not have the TK power and so would be unable to interact with a machine remotely. 

So if they want to interact with a machine, without being a silicon, they would have to have both DNA and the TK power.

Would the `!H.dna.check_mutation(TK)` cause an error if it were to run without `H` having DNA to have checked?

Reply:
As it is currently, if they don't have the tk power because they don't have dna or aren't a carbon then this proc will allow interaction even though they're failing `Adjacent(user)`

Reply:
Ah right. When I made this I assumed that mobs without DNA would never be able to interact with a machine anyway. So they would not even reach the proc.

That was probably not a great assumption to make. When I get back home I'll adjust it so mobs without DNA will be denied interaction as well. That would be safer.